src/coap_session.c

```diff
@@ -818,11 +819,22 @@ coap_session_create_client(
   coap_session_reference(session);
 
   if (proto == COAP_PROTO_UDP || proto == COAP_PROTO_DTLS) {
+    coap_session_t *s, *rtmp;
     if (!coap_socket_connect_udp(&session->sock, local_if, server,
       proto == COAP_PROTO_DTLS ? COAPS_DEFAULT_PORT : COAP_DEFAULT_PORT,
       &session->addr_info.local, &session->addr_info.remote)) {
       goto error;
     }
+    /* Check that this is not a duplicate 4-tuple */
+    SESSIONS_ITER_SAFE(ctx->sessions, s, rtmp) {
+      if ((s->proto == COAP_PROTO_UDP || s->proto == COAP_PROTO_DTLS) &&
+          memcmp(&session->addr_info, &s->addr_info,
```

Comment:
Not sure here: I would have explicitly called `coap_address_equals()` on `addr_info.remote` and `addr_info.local` to avoid issues if the platform-specific address implementation contained information that should not be compared (as seen earlier for byte-wise comparison of `sockaddr_in6`).

Reply:
Corrected and code pushed.